src/lib/logical_query_plan/data_dependencies/functional_dependency.cpp

```diff
@@ -137,8 +127,8 @@ FunctionalDependencies deflate_fds(const FunctionalDependencies& fds) {
   }
 
   auto deflated_fds = FunctionalDependencies(fds.size());
-  for (const auto& [determinants, dependents] : existing_fds) {
-    deflated_fds.emplace(determinants, dependents);
+  for (auto [determinants, dependents] : existing_fds) {
+    deflated_fds.emplace(std::move(determinants), std::move(dependents));
```

Comment:
You probably have a better  feeling of what is how often called. Is this the the dominating case? If so, should we rather copy a const ref to vector in the constructor?

Reply:
I'd say we do not query for dependencies too often, compared to, e.g., output expressions. Both would make a copy at the one or the other point, so I do not think it makes any difference.